src/test/java/de/filefighter/rest/cucumber/CommonCucumberSteps.java

```diff
@@ -1,34 +1,98 @@
+import de.filefighter.rest.domain.token.data.persistance.AccessTokenRepository;
+import de.filefighter.rest.domain.user.data.persistance.UserEntity;
+import de.filefighter.rest.domain.user.data.persistance.UserRepository;
 import io.cucumber.java.en.And;
 import io.cucumber.java.en.Given;
 import io.cucumber.java.en.Then;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.time.Instant;
+
+import static de.filefighter.rest.domain.token.business.AccessTokenBusinessService.ACCESS_TOKEN_DURATION_IN_SECONDS;
 
 public class CommonCucumberSteps extends CucumberIntegrationTest {
 
+    private final UserRepository userRepository;
+    private final AccessTokenRepository accessTokenRepository;
+    private final FileSystemRepository fileSystemRepository;
+
+    @Autowired
+    public CommonCucumberSteps(UserRepository userRepository, AccessTokenRepository accessTokenRepository, FileSystemRepository fileSystemRepository) {
+        this.userRepository = userRepository;
+        this.accessTokenRepository = accessTokenRepository;
+        this.fileSystemRepository = fileSystemRepository;
+    }
+
     @Given("database is empty")
     public void databaseIsEmpty() {
+        userRepository.deleteAll();
+        accessTokenRepository.deleteAll();
+        fileSystemRepository.deleteAll();
     }
 
     @And("user {long} exists")
     public void userExists(long userId) {
+        userRepository.save(UserEntity
+                .builder()
+                .userId(userId)
+                .build());
     }
 
     @And("user {long} has access token {string}")
     public void userHasAccessToken(long userId, String accessTokenValue) {
+        accessTokenRepository.save(AccessTokenEntity
+                .builder()
+                .userId(userId)
+                .value(accessTokenValue)
+                .validUntil(Instant.now().getEpochSecond() + ACCESS_TOKEN_DURATION_IN_SECONDS)
+                .build());
     }
 
     @And("user with id {long} exists and has username {string}, password {string} and refreshToken {string}")
     public void userWithIdExistsAndHasUsernamePasswordAndRefreshToken(long userId, String username, String password, String refreshTokenValue) {
+        userRepository.save(UserEntity
+                .builder()
+                .userId(userId)
+                .username(username)
+                .password(password)
+                .refreshToken(refreshTokenValue)
+                .build());
     }
 
     // file / folder
     @Given("{string} exists with id {long} and path {string}")
-    public void existsWithIdAndPath(String fileOrFolder, long fsItemId, String arg2) {
+    public void existsWithIdAndPath(String fileOrFolder, long fsItemId) {
+        if(fileOrFolder.equals("file")){
+            fileSystemRepository.save(FileSystemEntity
+                    .builder()
+                    .isFile(true)
+                    .id(fsItemId)
```

Comment:
why is path ignored?

Reply:
this is wip.

Reply:
nvm @qvalentin this is fixed one commit later :D